pkg/errutils/errutils.go

```diff
@@ -0,0 +1,14 @@
+package errutils
+
+import "github.com/pkg/errors"
+
+// WrapNil does the same as errors.Wrap but wraps even if e is nil
+func WrapNil(e, with error) error {
```

Comment:
This is a strange function. Why would a nil err be wrapped?

In the first use it seems like `err` is always nil, so this isn't necessary.

In the second use a `[]error` seems more appropriate, which can be joined together at the end to create a list of errors. (not sure if there is already a helper somewhere for building an error from a list of errors).

Wrapping errors is for building up the stack trace and context as it descends through abstractions. This is not how `WrapNil` is being used.

Reply:
Removed